### ISSUE TITLE
offset parameters not respected?

### ISSUE THREAD

**Issue by zachjweiner** [DTolm/VkFFT#53]
It appears to me that the buffer offset configuration parameters aren't being utilized: buffers always seem to be read from and/or written to at the head of the buffer. I'm testing on the OpenCL backend.

For example, say I make a buffer with double the length of the transform size and set only the second half to some input while setting the first half to zero. Passing the correct offset to the middle of the buffer yields an output with all zeros, while setting both halves of the array to the same input data yields the (correct) transform. Vice versa, outputs always seem to land at the head of the buffer regardless of passed offsets. I also peeked at the shader code and didn't see any of the offsets there (i.e., the literal number in items, not bytes).

Apologies that I don't have a simple reproducer to share; I'm working in Python and `pyvkfft` doesn't support offsets at the moment, so it's taken a bit of patching to attempt this. Let me know if I can provide any further information. Apologies in advanced if I'm missing something silly.

As an aside, is there something about the codegen that requires the offsets to be configuration parameters rather than launch parameters (i.e., arguments to the kernels themselves)? I imagine it'd be useful (or at least nice) to need only one VkFFTApplication for transforms that only differ by the offset of passed buffers.

**Comment by DTolm** [comment]
Hello,

You were correct, this was indeed a bug that occurred when buffers were provided at launch time. I have fixed it in the last version and added an option to provide offsets at launch time.

**Comment by zachjweiner** [comment]
Thanks for addressing this so quickly! (I should've thought to point out that I was indeed setting buffers only at launch time.) 

I'm now testing all combinations of zero and nonzero offsets for transforms, and I think I've found one bug. I observed that transforms that pass input and main buffers are incorrect when the input offset is zero but the main buffer offset is nonzero. I think [these two lines of the diff](https://github.com/DTolm/VkFFT/commit/4630e152ccb9934aef18653f6b16cef02ca6e22f#diff-d542fb4412375a2bd2319d9a25137c9fd850c5698e6ea47ec7f547b6e9b578eaR20973-R20974) are the culprit; commenting them out fixed this issue. Was there a separate reason for those offsets to be changed as such?
